pkg/signatures/public_key_verifier.go

```diff
@@ -219,7 +226,12 @@ func getVerifiedImageReference(signature oci.Signature, image *storage.Image) ([
 	var verifiedImageReferences []string
 	for _, name := range image.GetNames() {
 		if signatureImageReference == fmt.Sprintf("%s/%s",
-			name.GetRegistry(), name.GetRemote()) {
+			// Special case for the default registry we use internally for docker hub:
+			// We use docker.io, whilst the cosign / go-containerregistry library uses index.docker.io.
+			// Ensuring we get a match for the default docker registry, we conditionally replace the registry name
+			// here.
+			ternary.String(name.GetRegistry() == defaultDockerRegistry, cosignDockerHubRegistry,
```

Comment:
nit: perhaps we can stop using `ternary.String`/`ternary.Int` and can start using [IfThenElse](https://github.com/stackrox/stackrox/blob/0529062eb7396c82871e5f59305108638c7b27c0/pkg/utils/if_then_else.go#L4-L4)

Reply:
Obsolete with the changes made now, but good point I shall use it in the future 😄